packages/techdocs-common/src/stages/publish/types.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Entity, EntityName } from '@backstage/catalog-model';
+import express from 'express';
+
+/**
+ * Key for all the different types of TechDocs publishers that are supported.
+ */
+export type PublisherType = 'local' | 'google_gcs';
+
+export type PublisherBaseParams = {
+  entity: Entity;
+  /* The Path to the directory where the generated files are stored. */
+  directory: string;
+};
+
+export type PublisherBaseReturn = Promise<{}>;
+
+/**
+ * Base class for a TechDocs publisher (e.g. Local, Google GCS Bucket, AWS S3, etc.)
+ * The publisher handles publishing of the generated static files after the prepare and generate steps of TechDocs.
+ * It also provides APIs to communicate with the storage service.
+ */
+export interface PublisherBase {
+  /**
+   * Store the generated static files onto a storage service (either local filesystem or external service).
+   *
+   * @param options Object containing the entity from the service
+   * catalog, and the directory that contains the generated static files from TechDocs.
+   */
+  publish(options: PublisherBaseParams): PublisherBaseReturn;
```

Comment:
just an idea (i know everybody loves naming debates!): 

```ts
publish(request: PublishRequest): Promise<PublishResponse>;
```

(i've been liking to use the request/response pattern lately).

I think it's clearer to have the Promise visible in the signature because that's kinda what you expect

Reply:
> I think it's clearer to have the Promise visible in the signature because that's kinda what you expect

Agreed, thanks!